### PR TITLE
Avoid error when processing documents impacted by #3692

### LINKED ISSUE #3700: Automate cleanup for docs impacted by issue 3692
Simplify cleanup of datasets impacted by #3692 by ignoring (and potentially purging) non-leaf revision tree entries with leading non-JSON characters when encountered.

### PR DESCRIPTION
**Pull request by adamcfraser**
Fixes #3700.

### CODE REVIEW

**Comment by adamcfraser** [comment]
Currently includes commit from https://github.com/couchbase/sync_gateway/pull/3768 - need to rebase once that's merged.

- [ ] http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-integration-master/799/
- [ ] http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-integration-master/800/

File: db/crud_test.go [hunk 1]
```diff
@@ -352,3 +353,333 @@ func TestRevisionStoragePruneTombstone(t *testing.T) {
 	assertTrue(t, base.IsKeyNotFoundError(db.Bucket, err), "Revision should be not found")
 
 }
+
+// Checks for unwanted interaction between old revision body backups and revision cache
+func TestOldRevisionStorage(t *testing.T) {
+
+	db, _ := setupTestDBWithCacheOptions(t, CacheOptions{})
```
Comment (bbrks) [inline db/crud_test.go]: We'll need to defer `Close()` on the returned bucket to avoid test issues. I missed this in #3768

File: db/crud_test.go [hunk 2]
```diff
@@ -352,3 +353,333 @@ func TestRevisionStoragePruneTombstone(t *testing.T) {
...
+         "cas":"",
+         "value_crc32c":"",
+         "time_saved":"2018-09-27T13:47:44.719971735-07:00"
+     },
+     "key1":"value2",
+     "v":"6a"
+    }`
+
+func TestMalformedRevisionStorageRecovery(t *testing.T) {
+	db, _ := setupTestDBWithCacheOptions(t, CacheOptions{})
```
Comment (bbrks) [inline db/crud_test.go]: Again, need to defer `Close()` on the returned bucket

**Review by bbrks — CHANGES_REQUESTED** [review]
Couple of test teardown issues that need adding after rebasing

**Review by bbrks — APPROVED** [review]

**Pull request merged**